protocol/notifications.proto

```diff
@@ -0,0 +1,320 @@
+syntax = "proto3";
+package river;
+option go_package = "github.com/river-build/river/core/node/protocol";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+service AuthenticationService {
+  rpc StartAuthentication(StartAuthenticationRequest) returns (StartAuthenticationResponse);
+
+  rpc FinishAuthentication(FinishAuthenticationRequest) returns (FinishAuthenticationResponse);
+}
+
+message StartAuthenticationRequest {
+  // user id of the user that wants to authenticate
+  bytes user_id = 1;
+}
+
+message StartAuthenticationResponse {
+  // user id of the user that wants to authenticate
+  bytes user_id = 1;
+  // challenge that must be signed.
+  // clients must verify before signing that the challenge is >= 128 bits
+  bytes challenge = 2;
+  // indication until when challenge is valid
+  google.protobuf.Timestamp expiration = 3;
+}
+
+// note, challenge expiration is stored in the notification service and therefore doesn't have to be provided
+// in the request. Expiration is looked up through the nonce.
+message FinishAuthenticationRequest {
+  // user id of the user that wants to authenticate
+  bytes user_id = 1;
+  // challenge random nonce that was used to sign
+  bytes challenge = 2;
+  // signature calculated with:
+  // ecdsa_sign(client_key, sha256(PREFIX || user_id || expiration || challenge)
+  // with
+  // PREFIX=NS_AUTH: (string with length 8)
+  // user_id raw user id bytes (160 bits)
+  // expiration is a 64 bits unix timestamp encoded in big endian
+  // || means concatenation
+  bytes signature = 3;
```

Comment:
Can we add an optional delegate_sig here like we do in the StreamEvent? It's not vitally important, we can sign things with privy without throwing a modal, but it would be nice to have parity.

Reply:
We likely can but I have to look into the delegate signing part and figure out how it exactly works.